### ISSUE TITLE
Create tool that documents what flows over the IDD

### ISSUE THREAD

**Issue by akrherz** [Unidata/unidata-usercomm#14]
My blue sky volunteer project to demonstrate by the Spring 2018 meeting is a website that documents what flows over the IDD.  My initial functionality wish list for hacking at

- [ ] a realtime python process that databases LDM product metadata via a not well known feature that LDM has
- [ ] build out cross reference tables for feedtypes, WMO headers, awips IDs, to something human readable, etc
- [ ] build some aggregation stats to document the receipt of products such that the raw product flow information need not be kept in the database for too long
- [ ] a website that allows searching , viewing of the product receipts
- [ ] a website that complains about products that I literally have no idea what they are, so to shame various folks of interest :)

Will make some proof of concept and advertise on the `ldm-users` email list.

**Comment by semmerson** [comment]
One can specify in a `pqact(1)` configuration-file an action to be performed on a data-product that hasn't matched any previous entries. This ability might be useful. For more information, see the description of the `^_ELSE_$` pattern at [http://www.unidata.ucar.edu/software/ldm/ldm-current/basics/pqact.conf.html#introduction](http://www.unidata.ucar.edu/software/ldm/ldm-current/basics/pqact.conf.html#introduction)

**Comment by akrherz** [comment]
Hi @semmerson, thanks for your comment.  I was more referencing the `-metadata` and the somewhat undocumented `-nodata` PIPE options to stream all received metadata to my async python ingest.  The idea being:
```
ANY .*
  PIPE -nodata -metadata python fancypants.py
```
If you think this path of attack would end in tears, it would be good to know now :)

**Comment by semmerson** [comment]
@akrherz No tears. Just work.

**Comment by akrherz** [comment]
True to academic form, I wait for the last minute, well week before, to start working on the code for this project.  I now have a demonstration here with bare minimal functionality and also a backend archive that only starts at 18z 10 October 2017

https://id3b.agron.iastate.edu/

Will see how much iterative improvement I can do on this prior to group discussion at Unidata next week.  My github repo is [akrherz/id3b](https://github.com/akrherz/id3b).  If interested folks want to provide feedback on this GH issue or over on my GH repo as new issues, that would be great.  Otherwise, see you all in Boulder next week!

**Comment by semmerson** [comment]
Way cool!

One thing, though. I entered "KNES" into the "WMO Souce" search-box, while
leaving the other search-boxes blank, and got a long listing of products
that didn't have "KNES" in their source field.

Regards,
Steve Emmerson

**Comment by akrherz** [comment]
@semmerson thanks for checking it out and sorry about that bug.  The async javascript requests were stacking up with an older request returning results over-top the latest request, fixed with akrherz/id3b@9df5e11

I should also note that this is based on my IDD node of `metfs1.agron.iastate.edu`, which does not request `ANY ".*"` from Unidata.  It gets most products non-GOES16 and CONDUIT though.  I have noted that on the page now as well.

**Comment by akrherz** [comment]
So I had nightmares about this implementation last night with the conclusion being that the interface was of not much use unless the actual product data was available within the same interface.  One of the main uses I envisioned for the interface was to do some quality control work on the WMO headers for text and Grib products.  Without the products, this QC work would be even more tedious as one would have to cross check other data systems.  In a previous life, I had access to a NWS TOC system which had somewhat similar functionality to this app and having the raw text there was very useful.

Will have to boggle this some more before attempting implementation as doing such would explode compute/storage/IO requirements.

**Comment by sgdecker** [comment]
Would storing only, say, the first 256 bytes after the WMO/AWIPS headers be sufficient for this purpose? (I haven't done the math to see how much even a truncated product would explode the IO.)

**Comment by akrherz** [comment]
Hi @sgdecker, thanks for the feedback. That's an interesting idea.  Here's a current accounting of feedtypes and byte sizes since my ingest started yesterday afternoon

 ldm_feedtype |  feedtype | count  |     Total bytes     
------------- | ----- | ---- | --------------
268435456 |  NEXRAD3 |   340 |    537,231,360
32 |    WMO | 1089 |   1,612,542,195
16384 |   NLDN |  1304 |      2,050,328
8192 | FT13 |    2350 |   2,743,484,710
2097152 |   NIMAGE | 4077 |   3,335,567,469
1073741824 | NEXRAD2 | 115840 |  13,039,979,755
4 |  DDPLUS | 608282 |  16,824,278,691
8388608 |  NGRID | 949247 | 180,756,372,326
4096 | FT12 |  955275 |  56,109,996,794
11 | shrug | 1035354 |   1,602,212,770
134217728 | FT27 | 2084424 |  46,299,589,644

For the grib data, I suspect truncating at 256 bytes would cause troubles?  Maybe I just do the 'text' products to start as it is low hanging fruit and likely of most interest.  More boggling

**Comment by sgdecker** [comment]
Yeah, I guess for GRIB files it would be more useful to store the output of sending the product through wgrib/wgrib2 instead of the first x bytes. Looks like for feedtype FT27 storing 256 bytes puts you at 500-1000 MB per day rather than the 46+ GB of all the data. Maybe those are all GRIB products, though.
